druid-shell/src/window.rs

```diff
@@ -419,6 +422,14 @@ impl WindowHandle {
     pub fn get_scale(&self) -> Result<Scale, Error> {
         self.0.get_scale().map_err(Into::into)
     }
+
+    pub fn drop_context(&self) -> Option<DropContext> {
```

Comment:
We should not let a `DropContext` escape. This should be bound by "some" lifetime. maybe a `with_drop_context` that takes a closure?

Reply:
that will make this unusable for other events, even when a drag drop is active 
say a timer event

Reply:
Is there a particular case you had in mind, here?

Reply:
Earlier, it was `set_action` based on key events of modifier, but now modifiers are just available in all `drop_*` functions

Reply:
if `DropContext` is passed in various call:
say `fn drop_move(ctx: &'a DropContext, event: MouseEvent)`
now this is bound by a lifetime. means druid can not store this in its window handler. So `DropContext` will have to be part of `Event` which means adding a lifetime parameter :shrug: